Cortex/cortex_module/cortex_edr_connector.py

```diff
@@ -0,0 +1,170 @@
+import time
+from datetime import datetime, timedelta, timezone
+from dateutil.parser import isoparse
+
+from functools import cached_property
+import orjson
+from typing import Optional, Dict, Any, Tuple, List
+
+from requests.exceptions import HTTPError
+from sekoia_automation.connector import DefaultConnectorConfiguration
+from sekoia_automation.storage import PersistentJSON
+from urllib3.exceptions import HTTPError as BaseHTTPError
+
+from cortex_module.base import CortexConnector
+from cortex_module.client import ApiClient
+from cortex_module.metrics import EVENTS_LAG, FORWARD_EVENTS_DURATION, INCOMING_EVENTS, OUTCOMING_EVENTS
+
+
+class CortexEDRConfiguration(DefaultConnectorConfiguration):
+    chunk_size: int = 100
+    frequency: int = 60
+
+
+class CortexQueryEDRTrigger(CortexConnector):
+    """
+    Cortex EDR Query reads the alerts messages exposed after quering the Cortex
+    API and forward it to the playbook run.
+    """
+
+    configuration: CortexEDRConfiguration
+
+    def __init__(self, *args: Any, **kwargs: Optional[Any]) -> None:
+        super().__init__(*args, **kwargs)
+        self.context = PersistentJSON("context.json", self._data_path)
+        self.query: Dict[str, Any] = {
+            "request_data": {
+                "filters": [{"field": "creation_time", "operator": "gte", "value": 0}],
+                "search_from": 0,
+                "search_to": 0,
+                "sort": {"field": "creation_time", "keyword": "desc"},
+            }
+        }
+        self._timestamp_cursor = ""
+
+    @property
+    def timestamp_cursor(self) -> int:
+        now = datetime.now(timezone.utc)
+
+        with self.context as cache:
+            timestamp_cursor_str = cache.get("timestamp_cursor")
+
+            if timestamp_cursor_str is None:
+                before_two_days = now - timedelta(days=2)
+                return int(before_two_days.timestamp())
```

Comment:
For fresh connector, a shorter timedelta (like 5 minutes) is preferable. Otherwise, customers will think that the connector doesn't work as they do see their last events?